### PR TITLE
Optical flow loader for Davis dataset

### PR DESCRIPTION
**Pull request by fral92**

### CODE REVIEW

File: dataset_loaders/videos/davis.py [hunk 1]
```diff
@@ -55,12 +55,19 @@ def filenames(self):
...
                  split=.75,
                  *args, **kwargs):
 
         self.which_set = which_set
+        self.optical_flow_type = optical_flow_type
+        self.optical_flow_path = None
         self.threshold_masks = threshold_masks
 
+        if optical_flow_type is not None:
+            if optical_flow_type not in ['Brox', 'Farn', 'LK', 'TVL1']:
```
Comment (fvisin) [inline dataset_loaders/videos/davis.py:68]: Add `None` to the list and merge the two conditions.

File: dataset_loaders/videos/davis.py [hunk 2]
```diff
@@ -55,12 +55,19 @@ def filenames(self):
     def __init__(self,
                  which_set='train',
                  threshold_masks=False,
+                 optical_flow_type=None,
                  split=.75,
                  *args, **kwargs):
```
Comment (fvisin) [inline dataset_loaders/videos/davis.py:60]: Please add a description of the parameters (some of them you can copy from other datasets)

**Review by fvisin — CHANGES_REQUESTED** [review]
Thank you for the PR. Looks good overall, but I was wondering if you can make it more general so that we can share it with the other datasets.

How hard do you think it could be to move the relevant code in parallel_loader.py, where the data augmentation takes place? I don't have time to look into it in detail now, but I think it should be easy to check whether there is a dir in the dataset path containing the requested optical flow and in that case to load it from there rather than computing it.

From a high level perspective, this is what I would check:
- the optical flow should be computed on the unprocessed (uncropped / skewed / etc ..) image, so that we don't have to store all the possible transformations. I am afraid that we are doing the opposite at the moment (not to have to transform the OF), it should be checked what would be the downsides to do it the other way around
- define a standard way to name the directory and the images. The dir should be named after the technique and the chosen parameters.
- provide a flag to store the processed frames automatically (i.e., the first time we compute the OF and there is no local cache, if the flag is True we store the OF frames)

Do you think you have time to look into it and make these changes?

**Comment by marcociccone** [comment]
Please @fral92  close this because of #12

**Pull request closed without merging**